Sources/Ledger/PartialTxOut.swift

```diff
@@ -57,4 +58,21 @@ extension PartialTxOut {
             targetKey: targetKey,
             publicKey: publicKey)
     }
+    
+    private static func isCrc32Matching(reconstructed: Data32, txOutRecord: FogView_TxOutRecord) -> Bool {
+        let sentCommitment = txOutRecord.txOutAmountCommitmentData
+        let sentCrc32 = txOutRecord.txOutAmountCommitmentDataCrc32
+        let sentCommitmentCrc32: UInt32 = {
+            guard let sentCommitment32 = Data32(sentCommitment) else { return nil }
+            return TxOutUtils.calculateCrc32(from: sentCommitment32)
+        }() ?? .emptyCrc32
+        
+        let reconstructedCrc32 = TxOutUtils.calculateCrc32(from: reconstructed)
+        
+        return reconstructedCrc32 == sentCrc32 || reconstructedCrc32 == sentCommitmentCrc32
```

Comment:
Why do we use || instad of && here?

Reply:
because the crc32 isn't always sent in the protobuf

if it is sent then we have to reconstruct the commitment data from the other fields in the TxOut (and our account keys) and then calculate the crc32 from the reconstructed commitment for comparison.

If crc32 is not sent (current + soon to be deprecated) then we still reconstruct the commitment data and then calculate crc32 for sent & reconstructed and compare.

Only one side can be true, which is sufficient.